learning_resources_search/serializers.py

```diff
@@ -311,3 +330,25 @@ def serialize_for_deletion(opensearch_object_id):
         dict: the object deletion data
     """
     return {"_id": opensearch_object_id, "_op_type": "delete"}
+
+
+class OSLearningResourceSerializer(LearningResourceSerializer):
+    """
+    Serializer for indexing learning resources.
+    Adds custom fields needed for search purposes.
+    Any fields added here should also be added to SOURCE_EXCLUDED_FIELDS
+    """
+
+    SOURCE_EXCLUDED_FIELDS = [
+        "course.course_numbers.sort_coursenum",
+        "course.course_numbers.primary",
+    ]
+
+    def to_representation(self, instance):
+        """
+        Override to_representation to modify/add fields needed only for indexing
+        """
+        data = super().to_representation(instance)
+        if instance.resource_type == LearningResourceType.course.name:
+            add_extra_course_number_fields(data["course"]["course_numbers"])
```

Comment:
The document structure in opensearch will look like this, (with `primary` and `sorted_coursenum` excluded from hit results output):

```json
            "course": {
                "course_numbers": [
                    {
                        "value": "4.401",
                        "department": {
                            "name": "Architecture",
                            "department_id": "4"
                        },
                        "listing_type": "primary",
                        "primary": true,
                        "sort_coursenum": "04.401"
                    },
                    {
                        "value": "4.464",
                        "department": {
                            "name": "Architecture",
                            "department_id": "4"
                        },
                        "listing_type": "cross-listed",
                        "primary": false,
                        "sort_coursenum": "04.464"
                    }
                ]
            },
```
@abeglova  is this a suitable replacement for `department_course_numbers`?